src/storage.rs

```diff
@@ -68,23 +80,28 @@ pub struct RaftState {
 pub trait Storage {
     /// `initial_state` is called when Raft is initialized. This interface will return a `RaftState` which contains `HardState` and `ConfState`;
     fn initial_state(&self) -> Result<RaftState>;
+
     /// Returns a slice of log entries in the range `[low, high)`.
     /// max_size limits the total size of the log entries returned if not `None`, however
     /// the slice of entries returned will always have length at least 1 if entries are
     /// found in the range.
     fn entries(&self, low: u64, high: u64, max_size: impl Into<Option<u64>>) -> Result<Vec<Entry>>;
+
     /// Returns the term of entry idx, which must be in the range
     /// [first_index()-1, last_index()]. The term of the entry before
     /// first_index is retained for matching purpose even though the
     /// rest of that entry may not be available.
     fn term(&self, idx: u64) -> Result<u64>;
-    /// Returns the index of the first log entry that is
-    /// possible available via entries (older entries have been incorporated
-    /// into the latest snapshot; if storage only contains the dummy entry the
-    /// first log entry is not available).
+
+    /// Returns the index of the first log entry that is possible available via entries.
+    /// If the `Storage` is just initialized with a snapshot, `first_index` should return
+    /// the `snapshot index + 1`.
     fn first_index(&self) -> Result<u64>;
```

Comment:
Can we instruct what occurs when only the dummy entry is available?

Reply:
Here I think we don't need to mention `dummy entry` because it's public document of `Storage`. Dummy entry is just a implementation detail of `MemStorage`.

Reply:
OK.